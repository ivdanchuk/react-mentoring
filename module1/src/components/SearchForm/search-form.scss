.search-container {
  display: flex;
  align-items: center;
}

.search-container input {
  margin-right: 10px;
  background-color: #424242;
  color: white;
  border: none;
}

.search-container button {
  background-color: #f65261;
  border: none;
  color: white;
}
