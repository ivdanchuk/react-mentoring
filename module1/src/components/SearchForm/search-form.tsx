import React, { ChangeEvent, useState } from "react";

import styles from "./search-form.module.scss";

interface ISearchProps {
  initialQuery: string;
  onSearch: Function;
}

export const SearchForm = ({ initialQuery, onSearch }: ISearchProps) => {
  const [query, SetQuery] = useState(initialQuery || "Search movie ...");

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    SetQuery(event.target.value);
  };

  const handlePressEnter = ({ key }: React.KeyboardEvent) => {
    if (key === "Enter") {
      handleSearch();
    }
  };

  const handleSearch = () => {
    onSearch && onSearch(query);
  };

  return (
    <div className={styles.searchContainer}>
      <div className={styles.inputSearch}>
        <input
          type="text"
          value={query}
          onChange={(e) => handleInputChange(e)}
          onKeyDownCapture={(e) => handlePressEnter(e)}
        />
      </div>
      <div className={styles.searchButton} onClick={() => handleSearch}>
        SEARCH
      </div>
    </div>
  );
};
