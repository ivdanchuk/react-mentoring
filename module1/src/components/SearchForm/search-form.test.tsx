import {render, screen, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {SearchForm} from './search-form';

const mockOnSearch = jest.fn();
const initialText = 'text';
const changedText = 'new text';

const mockProps = {
    initialQuery: initialText,
    onSearch: mockOnSearch
}
const Tested = () => <SearchForm {...mockProps} />;

describe("<SearchForm/>", () => {

    test("checks initial props", () => {
        const {container} = render(<Tested/>);
        expect(screen.getByDisplayValue(initialText)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    test("checks input change", () => {
        const {container} = render(<Tested/>);
        const input = screen.getByDisplayValue(initialText);
        fireEvent.change(input, {target: {value: changedText}});
        expect(input).toHaveValue(changedText);
        expect(container).toMatchSnapshot();
    });

    test("checks search on button click", () => {
        const {container} = render(<Tested/>);
        const input = screen.getByDisplayValue(initialText);
        const button = screen.getByText(/SEARCH/i);
        fireEvent.change(input, {target: {value: changedText}});
        fireEvent.click(button);
        expect(mockOnSearch).toHaveBeenCalledWith(changedText);
        expect(container).toMatchSnapshot();
    });

    it("checks search on enter key press", () => {
        const {container} = render(<Tested/>);
        const input = screen.getByDisplayValue(initialText);
        fireEvent.change(input, {target: {value: changedText}});
        fireEvent.keyDown(input, {keyCode: 13});
        expect(mockOnSearch).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
    });
});