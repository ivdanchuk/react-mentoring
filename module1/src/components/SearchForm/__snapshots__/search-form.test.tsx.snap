// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm/> checks initial props 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      type="text"
      value="text"
    />
    <button>
      SEARCH
    </button>
  </div>
</div>
`;

exports[`<SearchForm/> checks input change 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      type="text"
      value="new text"
    />
    <button>
      SEARCH
    </button>
  </div>
</div>
`;

exports[`<SearchForm/> checks search on button click 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      type="text"
      value="new text"
    />
    <button>
      SEARCH
    </button>
  </div>
</div>
`;

exports[`<SearchForm/> checks search on enter key press 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      type="text"
      value="new text"
    />
    <button>
      SEARCH
    </button>
  </div>
</div>
`;
